src/OnlineSales/Tasks/EmailLogsToActivityLogDumpTask.cs

```diff
@@ -0,0 +1,75 @@
+﻿// <copyright file="EmailLogsToActivityLogDumpTask.cs" company="WavePoint Co. Ltd.">
+// Licensed under the MIT license. See LICENSE file in the samples root for full license information.
+// </copyright>
+
+using Microsoft.Extensions.Configuration;
+using OnlineSales.Configuration;
+using OnlineSales.Data;
+using OnlineSales.DTOs;
+using OnlineSales.Entities;
+using OnlineSales.Exceptions;
+using OnlineSales.Helpers;
+using OnlineSales.Services;
+using OnlineSales.Tasks;
+using Serilog;
+
+namespace OnlineSales.SendGrid.Tasks
+{
+    public class EmailLogsToActivityLogDumpTask : BaseTask
```

Comment:
Now we have three very similar task (SyncActivityLogTask in SendGrid, the same in Sms and here). Please consider creating the common base class. (I don't insist, may be there are some pitfalls, but it could be useful).

Reply:
I think there is no need of this because right now we have only 1 task in core which actually dumps things to activity log. And as for plugins. Every situation is different and different data requires different kinds of acquiring and transforming that's an edge case that the both plugins have dupe code. And futher ones create by/for community most likely be different. I could be wrong of course